spec.html

```diff
@@ -982,6 +982,7 @@ <h1>The String Type</h1>
         <p>The rationale behind this design was to keep the implementation of Strings as simple and high-performing as possible. If ECMAScript source text is in Normalized Form C, string literals are guaranteed to also be normalized, as long as they do not contain any Unicode escape sequences.</p>
       </emu-note>
       <p>In this specification, the phrase "the <dfn>string-concatenation</dfn> of _A_, _B_, ..." (where each argument is a String value, a code unit, or a sequence of code units) denotes the String value whose sequence of code units is the concatenation of the code units (in order) of each of the arguments (in order).</p>
+      <p>The phrase "the <dfn id="string-slice">string-slice</dfn> of _S_ from _inclusiveStart_ to _exclusiveEnd_" (where _S_ is a String value or a sequence of code units and _inclusiveStart_ and _exclusiveEnd_ are integers) denotes the String value consisting of the consecutive code units of _S_ beginning at index _inclusiveStart_ and ending immediately before index _exclusiveEnd_ (which is the empty String when _inclusiveStart_ = _exclusiveEnd_). If the "to" suffix is omitted, the length of _S_ is used as the value of _exclusiveEnd_.</p>
```

Comment:
why use prose for this instead of an abstract operation?

Reply:
I kind of prefer this prose. We could make string concatenation an AO as well, but for something so straightforward, it seems like overkill.

Reply:
Just following suit with string-concatenation and (to a lesser extent) [mathematical operations](https://tc39.es/ecma262/#sec-mathematical-operations) like abs/min/max/modulo. Do you feel strongly about it being a more formal abstract operation instead? And if so, where would it go? A new subsection under [7 Abstract Operations](https://tc39.es/ecma262/#sec-abstract-operations)?

Reply:
(I considered making string-concatenation an AO back when I introduced it. One reason I didn't was probably that it was unclear how to allow the bullet-list invocations (e.g., in [Number::toString](https://tc39.es/ecma262/#sec-numeric-types-number-tostring)) as an alternative to the standard prefix-parenthesized format.)

Reply:
String concatenation seems like a simple operation to me; slicing seems more complex and thus would benefit from being an AO.

If we made an AO, I'd expect it would indeed have to be a 7.5, or defined here, under the String Type.

@bakkot @syg do you have any thoughts about prose vs an AO for this?

Reply:
I'm happier with prose, personally.